packages/server/src/core/internals/config.ts

```diff
@@ -23,40 +21,62 @@ export const isServerDefault: boolean =
   !!process.env.JEST_WORKER_ID;
 
 /**
- * The initial params that are used and actually represents real values underneath
+ * The runtime config that are used and actually represents real values underneath
  * @internal
  */
-export interface InitOptions<TType extends InitGenerics> {
-  transformer: DataTransformerOptions;
-  errorFormatter: ErrorFormatter<TType['ctx'], any>;
+export interface RuntimeConfig<TTypes extends RootConfigTypes> {
+  /**
+   * Use a data transformer
+   * @link https://trpc.io/docs/data-transformers
+   */
+  transformer: TTypes['transformer'];
+  /**
+   * Use custom error formatting
+   * @link https://trpc.io/docs/error-formatting
+   */
+  errorFormatter: ErrorFormatter<TTypes['ctx'], any>;
   /**
    * Allow `@trpc/server` to run in non-server environments
+   * @warning **Use with caution**, this should likely mainly be used within testing.
    * @default false
    */
   allowOutsideOfServer: boolean;
   /**
    * Is this a server environment?
+   * @warning **Use with caution**, this should likely mainly be used within testing.
+   * @default typeof window === 'undefined' || 'Deno' in window || process.env.NODE_ENV === 'test'
    */
   isServer: boolean;
+  /**
+   * Is this development?
+   * Will be used to decide if the API should return stack traces
+   * @default process.env.NODE_ENV !== 'production'
+   */
+  isDev: boolean;
 }
 
 /**
  * @internal
  */
-export type CreateInitGenerics<TType extends InitGenerics> = TType;
+export type CreateRootConfigTypes<TGenerics extends RootConfigTypes> =
+  TGenerics;
 
 /**
  * The config that is resolved after `initTRPC.create()` has been called
  * Combination of `InitTOptions` + `InitGenerics`
  * @internal
  */
-export interface RootConfig extends InitGenerics {
-  transformer: CombinedDataTransformer;
-  // FIXME this should probably be restricted
-  errorShape: any;
+export interface RootConfig<TGenerics extends RootConfigTypes>
+  extends RuntimeConfig<TGenerics> {
+  $types: TGenerics;
 }
 
 /**
  * @internal
  */
-export type CreateRootConfig<TConfig extends RootConfig> = TConfig;
+export type AnyRootConfig = RootConfig<{
+  ctx: any;
+  meta: any;
```

Comment:
Would be slightly nicer if these were


```suggestion
  ctx: Record<any, any>;
  meta: Record<any, any>;
```